pluma/pluma-notebook.c

```diff
@@ -396,31 +396,37 @@ static void
 drag_start (PlumaNotebook *notebook,
 	    guint32        time)
 {
+	GdkSeat    *seat;
+	GdkDevice  *device;
+	GdkDisplay *display;
+
+	display = gtk_widget_get_display (GTK_WIDGET (notebook));
+	seat = gdk_display_get_default_seat (display);
+	device = gdk_seat_get_pointer (seat);
+
 	if (!leftdown) return;
 
 	notebook->priv->drag_in_progress = TRUE;
 
 	/* get a new cursor, if necessary */
 	/* FIXME multi-head */
 	if (cursor == NULL)
-	{
-		GdkDisplay *display;
-		display = gtk_widget_get_display (GTK_WIDGET (notebook));
 		cursor = gdk_cursor_new_for_display (display, GDK_FLEUR);
-	}
 
 	/* grab the pointer */
 	gtk_grab_add (GTK_WIDGET (notebook));
 
 	/* FIXME multi-head */
-	if (!gdk_pointer_is_grabbed ())
+	if (!gdk_display_device_is_grabbed (display, device))
 	{
-		gdk_pointer_grab (gtk_widget_get_window (GTK_WIDGET (notebook)),
-				  FALSE,
-				  GDK_BUTTON1_MOTION_MASK | GDK_BUTTON_RELEASE_MASK,
-				  NULL, 
-				  cursor, 
-				  time);
+		gdk_seat_grab (seat,
+			       gtk_widget_get_window (GTK_WIDGET (notebook)),
+			       GDK_SEAT_CAPABILITY_POINTER,
+			       FALSE,
+			       cursor,
+			       NULL,
```

Comment:
Hmm, `time` argument in `drag_start` function is now unused. We can either drop it from the function, or replace it with a `GdkEvent`, which can be passed into `gdk_seat_grab`.

Reply:
ok, please, feel free to do the job :)

Reply:
Sure, I'll make changes.

Reply:
There we go, https://github.com/mate-desktop/pluma/pull/326